tests/controls/windowtest.cpp

```diff
@@ -148,6 +149,16 @@ TEST_CASE_METHOD(WindowTestCase, "Window::Mouse", "[window]")
 
     CHECK(m_window->GetCursor().IsOk());
 
+    wxCursor resCursor1( "horse" );
+    CHECK( resCursor1.IsOk() );
+#ifdef __WXOSX__
+    wxRemoveFile( wxStandardPaths::Get().GetResourcesDir() + "horse.png" );
```

Comment:
Why do you call `wxRemoveFile()` here??

Reply:
@vadz ,
The code checks for presence of png and cur file.
I need to check if both cases are covered as well as "no file present"

So I need to remove png and then cur file..

Or you mean it is easier to use simple "rm" here?

Reply:
I mean that you really must explain about what you're doing. You have some very original, to put it mildly, ideas about how tests should work and the least you can do is to carefully document your thinking because I assure you it's not obvious.

In this particular case, it seems -- to anybody else except you, apparently -- obvious that the test shouldn't modify the file system in any permanent way so it should never delete any files not created by it.

It also seems very unclear what exactly is the test testing for because it still expects to create the cursor successfully after removing the file (probably because `wxRemoveFile()` fails because you don't pass it the correct path...).

In short, I have absolutely no idea about what are you trying to do here but I'm pretty sure that what you currently do is wrong.

Please explain in the comments here what is the code supposed to do and/or better think more about this and change it.

Reply:
@vadz ,
I added the comment explaining the reason.

Let me know if this is not enough.